hazelcast/src/main/java/com/hazelcast/topic/impl/TotalOrderedTopicProxy.java

```diff
@@ -32,19 +40,68 @@
  */
 public class TotalOrderedTopicProxy<E> extends TopicProxy<E> {
 
+    private final OperationService operationService;
     private final int partitionId;
+    private final LocalTopicStatsImpl topicStats;
 
     public TotalOrderedTopicProxy(String name, NodeEngine nodeEngine, TopicService service) {
         super(name, nodeEngine, service);
-        this.partitionId = nodeEngine.getPartitionService().getPartitionId(getNameAsPartitionAwareData());
+        this.partitionId = nodeEngine.getPartitionService()
+                .getPartitionId(getNameAsPartitionAwareData());
+        this.operationService = nodeEngine.getOperationService();
+        this.topicStats = service.getLocalTopicStats(name);
     }
 
     @Override
     public void publish(@Nonnull  E message) {
         checkNotNull(message, NULL_MESSAGE_IS_NOT_ALLOWED);
-        Operation operation = new PublishOperation(getName(), toData(message))
-                .setPartitionId(partitionId);
+        Operation operation = new PublishOperation(getName(), toData(message)).setPartitionId(partitionId);
         InternalCompletableFuture f = invokeOnPartition(operation);
         f.joinInternal();
     }
+
+    @Override
+    public InternalCompletableFuture<Void> publishAsync(@Nonnull E message) {
+        checkNotNull(message, NULL_MESSAGE_IS_NOT_ALLOWED);
+        Operation op = new PublishOperation(getName(), toData(message));
+        return publishInternalAsync(op);
+    }
+
+    @Override
+    public void publishAll(@Nonnull Collection<? extends E> messages) {
+        publishAllAsync(messages).joinInternal();
+    }
+
+    @Override
+    public InternalCompletableFuture<Void> publishAllAsync(@Nonnull Collection<? extends E> messages) {
+        checkNotNull(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
+        checkNoNullInside(messages, NULL_MESSAGE_IS_NOT_ALLOWED);
+        // RU_COMPAT_4_0
+        checkClusterVersion(Versions.V4_1);
+
+        Operation op = new PublishAllOperation(getName(), toDataArray(messages));
+        return publishInternalAsync(op);
+    }
+
+    private InternalCompletableFuture<Void> publishInternalAsync(Operation operation) {
+        try {
+            final InvocationFuture<Void> invocationFuture = operationService.invokeOnPartition(
+                    OperationService.SERVICE_NAME, operation, partitionId);
+            topicStats.incrementPublishes();
```

Comment:
Looks like stats are incremented in the operation so you can remove it from here.

Reply:
Done